schema-registry/rest-service/src/test/java/com/hortonworks/registries/schemaregistry/avro/AbstractAvroSchemaRegistryCientTest.java

```diff
@@ -44,20 +44,28 @@
  */
 public abstract class AbstractAvroSchemaRegistryCientTest {
 
-    @ClassRule
-    public static final DropwizardAppRule<RegistryConfiguration> DROPWIZARD_APP_RULE
-            = new DropwizardAppRule<>(RegistryApplication.class, ResourceHelpers.resourceFilePath("schema-registry-test.yaml"));
-
-    protected final String rootUrl = String.format("http://localhost:%d/api/v1", DROPWIZARD_APP_RULE.getLocalPort());
-    protected final Map<String, String> SCHEMA_REGISTRY_CLIENT_CONF = Collections.singletonMap(SchemaRegistryClient.Configuration.SCHEMA_REGISTRY_URL.name(), rootUrl);
+    protected static LocalSchemaRegistryServer localSchemaRegistryServer;
```

Comment:
All static fields should be made  instance fields and they maintain state for the given `SchemaRegistryTestProfileType`.  So, it is better to have constructor which takes `SchemaRegistryTestProfileType` and you can also have respective `close` method which can have the same logic as current teardown method.

Other methods which are generating avro payloads for different avro schemas can be extracted into avro generator class for the given schemas.